util/process_triggers_folder.ts

```diff
@@ -44,35 +46,57 @@ const changeExportToPush = (lines: string[]) => {
   return lines;
 };
 
-const lintToFile = async (lines: string[], outputFileName: string) => {
+const lint = async (lines: string[]) => {
   const linter = new eslint.ESLint({ fix: true });
   const contents = lines.join('\n');
   // Deliberately don't pass filename here, as dist/ is ignored in eslint.
   const results = await linter.lintText(contents);
 
   // There's only one result from lintText, as per documentation.
-  const lintResult = results[0];
-  if (!lintResult) {
-    console.error('No result from linting?');
-    process.exit(2);
-  }
-  if (lintResult.errorCount > 0 || lintResult.warningCount > 0) {
-    // Errors aren't great, but tsc can create unfixable line length errors, so ignore them.
-    console.error(`Lint ran with errors: ${JSON.stringify(lintResult.messages)}`);
-  }
-
-  // Overwrite the file, if it already exists.
-  fs.writeFileSync(outputFileName, lintResult.output);
+  return results[0];
 };
 
 const processFile = async (filename: string) => {
-  console.log(`Processing file: ${filename}`);
   const originalContents = fs.readFileSync(filename).toString();
   let lines = originalContents.split(/[\r\n]+/);
 
   lines = removeImports(lines);
   lines = changeExportToPush(lines);
-  await lintToFile(lines, filename);
+  const lintResult = await lint(lines);
+  if (!lintResult) {
+    console.error('${filename}: No result from linting?');
+    process.exit(2);
+  }
+
+  if (lintResult.errorCount > 0) {
+    console.error(`${filename}: Lint ran with errors: ${JSON.stringify(lintResult.messages)}`);
+    process.exit(3);
+  }
+
+  if (lintResult.warningCount > 0) {
+    // Warnings aren't great, but tsc can create unfixable line length warnings.
+    // Print them, but don't stop.
+    console.error(`${filename}: Lint ran with warnings: ${JSON.stringify(lintResult.messages)}`);
+  }
+
+  const contents = lintResult.output;
+  if (!contents) {
+    console.error(`${filename}: Lint returned no contents`);
+    process.exit(4);
+  }
```

Comment:
Does output return an all-clear normally?

Reply:
What do you mean by that?

Reply:
If everything is good, we still expect an output?

Reply:
Yeah, the output is the linted file.  I guess technically this could be empty (and valid) vs undefined, so I should probably be more precise here.